main.go

```diff
@@ -17,10 +18,21 @@ func RetrieveSecret(variableName string) {
 	sess, err := session.NewSessionWithOptions(session.Options{
 		SharedConfigState: session.SharedConfigEnable,
 	})
-	if err != nil { // resp is now filled
+	if err != nil {
 		printAndExit(err)
 	}
 
+	// AWS Go SDK does not currently support automatic fetching of region from ec2metadata.
+	// If the region could not be found in an environment variable or a shared config file,
+	// create metaSession to fetch the ec2 instance region and pass to the regular Session.
+	if *sess.Config.Region == "" {
+		metaSession, _ := session.NewSession()
```

Comment:
Throwing away the error returned here and below (by `Region`) seems like it could make problems harder to diagnose. Is there any reason not to call `printAndExit` if error is non-nil?

Reply:
Failing silently here causes a more helpful error to be thrown later. 
If I `printAndExit` here or after fetching the region, it throws: `RequestError: send request failed
caused by: Get http://169.254.169.254/latest/meta-data/placement/availability-zone: dial tcp 169.254.169.254:80: connect: host is down` as opposed to `MissingRegion: could not find region configuration` when I don't catch this error.

Reply:
As an alternative, looks like you can use `EC2Metadata.Available` to determine whether you're on an EC2 instance. So, you could do this, instead:

```
	if *sess.Config.Region == "" {
		metaSession, err := session.NewSession()
		if err != nil {
			printAndExit(err)
		}
		metaClient := ec2metadata.New(metaSession)
		if metaClient.Available() {
			if region, err := metaClient.Region(); err == nil {
				sess.Config.Region = aws.String(region)
			} else {
				printAndExit(err)
			}
		}
	}
```